codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -341,6 +344,10 @@ public void error(Throwable cause) {
 
         @Override
         public void write(int allowedBytes) {
+            if (promise.isVoid()) {
```

Comment:
Should we just move the lazy creation of the `newPromise` to where we set `writePromise = promise`?